samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -334,7 +335,7 @@ object SamzaContainer extends Logging {
 
     val chooserFactory = Util.getObj[MessageChooserFactory](chooserFactoryClassName)
 
-    val chooser = DefaultChooser(inputStreamMetadata, chooserFactory, config, samzaContainerMetrics.registry, systemAdmins)
+    val chooser = DefaultChooser(inputStreamMetadata, chooserFactory, config, samzaContainerMetrics.registry, systemAdminMap)
```

Comment:
Not sure why we're passing the systemAdminMap in some places and systemAdmins in others. How bad is it to remove SystemAdmins?

Reply:
I can replace SystemAdminMap with SystemAdmins everywhere. It just requires more refactor and cosmetic change in this patch. Do you prefer me to do that?

Reply:
Fixed as discussed.